src/context.rs

```diff
@@ -673,17 +791,26 @@ impl Context {
     }
 
     /// Read the public part of a key currently in the TPM and return it.
-    pub fn read_public(&mut self, key_handle: ESYS_TR) -> Result<TPM2B_PUBLIC> {
+    pub fn read_public(&mut self, key_handle: KeyHandle) -> Result<TPM2B_PUBLIC> {
         let mut public = null_mut();
         let mut name = null_mut();
         let mut qualified_name = null_mut();
         let ret = unsafe {
             Esys_ReadPublic(
                 self.mut_context(),
-                key_handle,
-                self.sessions.0,
-                self.sessions.1,
-                self.sessions.2,
+                key_handle.into(),
+                self.sessions
+                    .0
+                    .map(|v| v.handle().into())
+                    .unwrap_or(ESYS_TR_NONE),
```

Comment:
Might be worth to create a method somewhere to do the conversion `Option<Session>` -> `ESYS_TR`.

Reply:
Hrrmm...maybe I am not sure what I think about that. I have to think about that.

Reply:
Note that my point is only to avoid code duplication as opposed as esthetics ;)

Reply:
That things are just aliases for an integer type is something you will probably never get away from when the code is written in C unless it is written by a zealous programmer with a fetish for types. So conversions to and from ESYS_TR is something that is done for most handles and session types.

To be able to do:
```rust
self.sessions.0.into()
```
instead of 
```rust
self.sessions.0.map(|v| v.handle().into()).unwrap_or(ESYS_TR_NONE)
```

So how it is used inside the Context methods is of little interest really because that is an implementation detail that should not concern users unless there is a bug. Something that is more interesting is how it would be used used by some one that is using the lib and then suddenly need to value of the handle.

OBS! Pseudo code.
```rust
// bla bla code
let mut auth_session: Option<Session> = None;
if need_to_auth {
    auth_session = context.start_auth_session()
}

let actual_value: ESYS_TR = auth_session.into();
```

That is how I would expect it to work if I didn't know better. Which speaks in favor for adding conversions from Option<Session>. So I think I may be inclined into adding that conversion.

Reply:
Thanks!! That sounds good

Reply:
Unfortunatly ```From<Option<Session>> for ESYS_TR``` is not possible. Because neither Option nor ESYS_TR are defined in the crate.

I have been going at this for a little while now and more I try the worse the code gets.

Reply:
Yeah, I didn't realise that initially! Thanks for attempting 😅